wolfcrypt/test/test.c

```diff
@@ -25012,10 +25012,73 @@ static int crypto_ecc_sign(const uint8_t *key, uint32_t keySz,
 
     return ret;
 }
+#endif /* HAVE_ECC_SIGN && HAVE_ECC_VERIFY */
+
+#ifdef HAVE_ECC_DHE
+/* 
+ * Currently we only have support for P-256/384 for non-blocking key gen and
+ * shared secret gen.
+ */
+static int ecc_test_nonblock_dhe(WC_RNG* rng) {
+    int ret = 0;
+    ecc_key keyA;
+    ecc_key keyB;
+    ecc_nb_ctx_t nbCtxA;
+    ecc_nb_ctx_t nbCtxB;
+    byte secretA[ECC_SHARED_SIZE];
+    byte secretB[ECC_SHARED_SIZE];
+    word32 secretSzA = ECC_SHARED_SIZE;
+    word32 secretSzB = ECC_SHARED_SIZE;
+
+    if (ret == 0) {
+        ret = wc_ecc_init(&keyA);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_init(&keyB);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_set_nonblock(&keyA, &nbCtxA);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_set_nonblock(&keyB, &nbCtxB);
+    }
+    if (ret == 0) {
+        do {
+            ret = wc_ecc_make_key_ex(rng, ECC_CURVE_SZ, &keyA, ECC_CURVE_ID);
+        } while (ret == FP_WOULDBLOCK);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_check_key(&keyA);
+    }
+    if (ret == 0) {
+        ret = wc_ecc_import_unsigned(&keyB, kPubKey, kPubKey + ECC_CURVE_SZ,
+                                     kPrivKey, ECC_CURVE_ID);
+    }
+    if (ret == 0) {
+        do {
+            ret = wc_ecc_shared_secret(&keyA, &keyB, secretA, &secretSzA);
```

Comment:
Same here, please add count.

Reply:
Please check:

```
`./configure --enable-ecc=nonblock --enable-sp=yes,nonblock CFLAGS="-DWOLFSSL_PUBLIC_MP -DWC_ECC_NONBLOCK_ONLY -DECC_USER_CURVES -DHAVE_ECC384" --enable-debug --disable-shared && make
./wolfcrypt/test/testwolfcrypt
...
ECC non-block sign: 25021 times
ECC non-block verify: 49019 times
ECC non-block key gen: 1 times
ECC non-block shared secret: 1 times
```

Also same results if I remove the `-DWC_ECC_NONBLOCK_ONLY`.

Reply:
Looks like I missed the changes to add non-blocking for P-521 to ecc.c. Thanks.